github/messages.go

```diff
@@ -117,3 +120,52 @@ func validateSignature(signature string, payload, secretKey []byte) error {
 	}
 	return nil
 }
+
+// getWebhookEvent parses the payload and request, and build the Event.
+func getWebhookEvent(r *http.Request, payload []byte) (interface{}, error) {
+	eventType := r.Header.Get(eventTypeHeader)
+	if eventType == "push" {
+		pushEvent := WebHookPayload{}
+		if err := json.Unmarshal(payload, &pushEvent); err != nil {
+			return nil, err
+		}
+		return pushEvent, nil
+	}
+
+	event := Event{
+		Type:       &eventType,
+		RawPayload: (*json.RawMessage)(&payload),
+	}
+	return event.Payload(), nil
```

Comment:
I don't think this is going to work for any of the other event types because the string returned in the header does not match the strings used in `Event.Payload`, correct?